uktrade/great-cms
# Redirect unverified user to verification code page after login

### Description
This PR makes an update to the login flow to make it easier for users who have previously signed up, but not yet verified their account, to login.

**Previous flow:**
- Unverified (exists in the db but hasn't entered a verification code) user enters their details into the login form
- They are redirected to `/dashboard`
- Because the dashboard view is protected by the `AuthenticatedUserRequired` mixin, they are then redirected back to `/signup`

**New flow:**
- Unverified user enters their details into the login form
- Directory SSO returns a 401 unauthorised response
- This status code triggers the re-issuing of a verification code email with a new code attached
- The `uidb64` and `verification_token` values are taken from the new code which is created, and returned in the response object
- The `handleSubmit` function in the client code determines the url to redirect to based on the existence of these response parameters
- The user is then redirected to the same URL which would be in the verification code email, e.g. `http://greatcms.trade.great:8020/signup/?uidb64=OA&token=636-5298c48e1e68b7946672`

### To test locally
- Run against the `feature/unverified-user-login-flow` branch in `directory-sso`
- Create a new user via `/signup`
- Do not verify that user
- Go to `/login` and enter the details for the new user
- Check that you are redirected to the verification code step of the signup flow

### Screenshots

![Screenshot 2022-08-04 at 11 58 55](https://user-images.githubusercontent.com/22460823/182831129-21cf0ab1-1bf7-407c-8e4e-8edfb48b4163.png)


### Workflow


### Reviewing help


### Housekeeping


### Merging
